### PR TITLE
Rewrite the page annotation to follow the Route annotation

### PR DESCRIPTION
**Pull request by aschempp**
| Q                | A
| -----------------| ---
| Fixed issues     | Fixes https://github.com/contao/contao/pull/1971

This is mostly about allowing a string for methods, same as Symfony does. There are two use cases:

```php
// Custom page controller
/**
 * @Page(methods="GET")
 */
class MyPageController
{
}
```

```yaml
// services.yml
services:
    App\Controller\Page\MyPageController:
        tags:
            - { name: 'contao.page', methods: 'GET' }
```

### CODE REVIEW

File: core-bundle/src/Routing/Page/RouteConfig.php [hunk 1]
```diff
@@ -45,19 +45,19 @@ final class RouteConfig
     private $defaults;
 
     /**
-     * @var array
+     * @var array<string>|string
```
Comment (leofeyer) [inline core-bundle/src/Routing/Page/RouteConfig.php]:
```suggestion
     * @var string|array<string>
```

File: core-bundle/src/ServiceAnnotation/Page.php [hunk 2]
```diff
@@ -22,32 +20,53 @@
...
+ *
+ * @see \Symfony\Component\Routing\Annotation\Route
  */
 final class Page implements ServiceAnnotationInterface
 {
-    /**
-     * @var array
-     */
-    private $attributes;
+    private $type;
```
Comment (leofeyer) [inline core-bundle/src/ServiceAnnotation/Page.php:31]: The class properties should have `@var` annotations.

**Review by leofeyer — CHANGES_REQUESTED** [review]

File: core-bundle/src/ServiceAnnotation/Page.php (hunk 2 above)
Comment (leofeyer) [reply core-bundle/src/ServiceAnnotation/Page.php:31]: (We could also abolish this rule in the future, but for now it still applies.)

Comment (leofeyer) [reply core-bundle/src/ServiceAnnotation/Page.php:31]: See #1974.

File: core-bundle/src/Routing/Page/RouteConfig.php (hunk 1 above)
Comment (aschempp) [reply core-bundle/src/Routing/Page/RouteConfig.php]: fixed in 7cc0debe1e3549a1743a89fb9a87475285d769b4. It's actually always an array, but the constructor argument is not.

File: core-bundle/src/ServiceAnnotation/Page.php (hunk 2 above)
Comment (aschempp) [reply core-bundle/src/ServiceAnnotation/Page.php:31]: added in 7cc0debe1e3549a1743a89fb9a87475285d769b4

**Pull request merged**